packages/sdk/src/query_on_connect/index.ts

```diff
@@ -0,0 +1,208 @@
+import { type PeerId, TypedEventEmitter } from "@libp2p/interface";
+import {
+  HealthStatus,
+  type IDecodedMessage,
+  type IDecoder,
+  IWakuEventEmitter,
+  QueryRequestParams,
+  WakuEvent
+} from "@waku/interfaces";
+import { Logger } from "@waku/utils";
+
+import {
+  IPeerManagerEvents,
+  PeerManagerEventNames
+} from "../peer_manager/peer_manager.js";
+
+const log = new Logger("sdk:query-on-connect");
+
+export const DEFAULT_FORCE_QUERY_THRESHOLD_MS = 5 * 60 * 1000; // 5 minutes
+export const MAX_TIME_RANGE_QUERY_MS = 24 * 60 * 60 * 1000; // 24 hours
+
+export interface QueryOnConnectOptions {
+  /**
+   * Elapsed time since the last successful query, after which we proceed with
+   * a store query, on a connection event, no matter the conditions.
+   * @default [[DEFAULT_FORCE_QUERY_THRESHOLD_MS]]
+   */
+  forceQueryThresholdMs?: number;
+}
+
+export enum QueryOnConnectEvent {
+  /**
+   * A message has been retrieved.
+   */
+  MessagesRetrieved = "messages:retrieved"
+}
+
+export type QueryOnConnectEvents = {
+  [QueryOnConnectEvent.MessagesRetrieved]: CustomEvent<IDecodedMessage[]>;
+};
+
+/**
+ * Proceed with time-range store queries after connection to a store node.
+ * Partial implementation of [Waku P2P Reliability](https://github.com/waku-org/specs/blob/master/standards/application/p2p-reliability.md)
+ *
+ * @emits <T extends IDecodedMessage> message retrieved on "messages"
+ */
+export class QueryOnConnect<
```

Comment:
Is this module used anywhere? Or you want to introduce it in this PR and then consume in follow up? 

note: I am implementing similar thing for Send API but was planning to add in the same PR where it would be used

Reply:
See description. To be used in reliable Channel. I split PR to make it easier to review.

Once this and reliable channel PRs merged, you are welcome to refactor. Would make sense to delete this once Waku API is done with p2p reliability and reliable channel uses Waku API.

Reply:
that works